src/SignalR/clients/java/signalr/test/src/main/java/com/microsoft/signalr/GsonHubProtocolTest.java

```diff
@@ -409,7 +417,52 @@ public void errorWhileParsingIncompleteMessage() {
         TestBinder binder = new TestBinder(new Type[] { int.class, int.class }, null);
 
         RuntimeException exception = assertThrows(RuntimeException.class,
-                () -> jsonHubProtocol.parseMessages(message, binder));
+                () -> hubProtocol.parseMessages(message, binder));
         assertEquals("Message is incomplete.", exception.getMessage());
     }
-}
\ No newline at end of file
+
+    @Test
+    public void invocationBindingFailureWhenParsingLocalDateTimeWithoutAppropriateTypeAdaptor() {
+        // Create message with LocalDateTime payload
+        String json = "{\"type\":1,\"target\":\"test\",\"arguments\":[\"2022-12-13T09:13:00\"]}\u001E";
+        ByteBuffer bytes = TestUtils.stringToByteBuffer(json);
+        TestBinder binder = new TestBinder(new Type[] { LocalDateTime.class }, null);
+
+        List<HubMessage> messages = hubProtocol.parseMessages(bytes, binder);
+        assertNotNull(messages);
+        assertEquals(1, messages.size());
+        HubMessage message = messages.get(0);
+        assertEquals(HubMessageType.INVOCATION_BINDING_FAILURE, message.getMessageType());
+        InvocationBindingFailureMessage failureMessage = (InvocationBindingFailureMessage) messages.get(0);
+
+        assertEquals("java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 41 path $.arguments[0]", failureMessage.getException().getMessage());
+    }
+
+    @Test
+    public void canParseLocalDatetimeWithAppropriateTypeAdaptor() {
+        LocalDateTime expectedResult = LocalDateTime.parse("2022-12-13T09:13:00");
+
+        // Setup appropriate type adaptor
+        Gson gson = new GsonBuilder()
+                .registerTypeAdapter(LocalDateTime.class, ((JsonDeserializer<LocalDateTime>) (json, type, context)
```

Comment:
Can Gson add custom `int` and `string` adapters? If so, we should add a test that `type`, `target`, and `invocationId` fields aren't affected by Gson config.

Reply:
You can register type adapters for literally any type, including int and string. The type adapters, however, do not affect the SignalR json protocol stuff, only its payloads.

I've added a test (1937b5f2002b84add8c36f750e36a98d305d031e) with a pretty rude type adapter that attempts to convert any `String` to an `int` using the pattern `"one"` -> `1`, `"two"` -> `2`, `"three"` -> `3`,  default throw exception.

`type`, `target`, and `invocationId` are completely unaffected by this.

Reply:
Does that only test the `type` field? Since the adapter will only run when trying to deserialize Integer and `target` and `invocationId` are strings.

Reply:
> The implementation of `parseMessages` is using manual json parsing (instead of mapping to POJOs) which is why the type adaptor stuff doesnt affect it

Yes, I'm very aware...
That doesn't mean we shouldn't have tests though, so it doesn't accidentally break in the future.

Reply:
Both `target` and `invocationId` are covered as of b3f97f3c40bbb974ee4e636b14d8366527957c9b. I've added yet another type adapter to the same test, such that ints are converted to strings, and strings are also converted to ints. (The same element won't be converted back and forth, don't worry)
This is a toy example, but I think it covers your concern.